src/main/java/edu/kit/kastel/sdq/artemis4j/client/FeedbackDTO.java

```diff
@@ -0,0 +1,54 @@
+/* Licensed under EPL-2.0 2024. */
+package edu.kit.kastel.sdq.artemis4j.client;
+
+import java.util.List;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import edu.kit.kastel.sdq.artemis4j.ArtemisNetworkException;
+
+/**
+ * Corresponds to an annotation as part of an assessment
+ * 
+ * @param type                type of the feedback
+ * @param id                  null for manual feedback
+ * @param credits             credits given to the student, may be negative
+ * @param positive            null for manual feedback
+ * @param visibility          null for manual feedback
+ * @param text                null for unreferenced manual & automatic feedback,
+ *                            of form "File
+ *                            src/edu/kit/informatik/BubbleSort.java at line 13"
+ * @param reference           null for unreferenced manual & automatic feedback,
+ *                            of form
+ *                            "file:src/edu/kit/informatik/BubbleSort.java_line:12"
+ * @param detailText          null for automatic feedback
+ * @param hasLongFeedbackText whether long feedback needs to be fetched for the
+ *                            detailText
+ * @param testCase            the associated test case, null for non-automatic
+ *                            feedback
+ */
+public record FeedbackDTO(@JsonProperty FeedbackType type, @JsonProperty Integer id, @JsonProperty double credits, @JsonProperty Boolean positive,
+		@JsonProperty String visibility, @JsonProperty String text, @JsonProperty String reference, @JsonProperty String detailText,
+		@JsonProperty Boolean hasLongFeedbackText, @JsonProperty TestCaseDTO testCase) {
```

Comment:
These attributes are `Boolean`, because they can be null? Consider making them `boolean` instead with a default value. The nullability will likely cause problems in the future. For example, the `hasLongFeedbackText` is used in `ProgrammingSubmissionDTO` in L68-70 without first checking for `null`:
```java
				if (feedback.hasLongFeedbackText()) {
					detailText = FeedbackDTO.fetchLongFeedback(client, result.id(), feedback.id());
				}
```
Not sure what happens when you try to unbox a `null` `Boolean`, but it is likely to crash?

Reply:
The same problem is with `Integer`. Alternatively, we could do `Optional<Integer>` instead to make the potential absence of the value explicit.

Reply:
I'd rather get some kind of crash instead of a silent error due to a default value if Artemis changes it's data representation. 

Regarding Optional: It's really only intended to be used as a return value (see e.g. its Jdoc and numerous articles online)

Reply:
From what I can tell, this notion is highly debated and not universally shared. I think an Optional conveys the meaning a lot better and because this is a record, you could say that they are return values of the generated getters.

Reply:
But we can leave it as is, just wanted to warn that this might be a problem in the future